url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/HttpResponse.java

```diff
@@ -0,0 +1,237 @@
+		return addHeader(new HttpRequestHeader(key, value));
+	}
+
+	public HttpResponse addHeader(HttpRequestHeader header) {
+//		System.out.println("key="+header.getKey()+","+header.getValue());
+		if (header.getKey().toLowerCase()
+				.equals(HttpRequestHeader.HEADER_CONTENT_LENGTH.toLowerCase())) {
+			mContentLength = header;
+		}
+		mHeaders.add(header);
+		return this;
+	}
+
+	public long getContentSizeFromHeader() {
+		if(mContentLength == null) {
+			return Integer.MAX_VALUE;
+		} 
+		String value = mContentLength.getValue();
+		try {
+			return Integer.parseInt(value.replaceAll(" ", ""));
+		} catch(Exception e) {
+			return 0;
+		}
+	}
+
+	@Override
+	public void encode(OutputStream output) throws IOException {
+		output.write((mHttpVersion + SP + mStatusCode + SP + mReasonPharse + CRLF)
+				.getBytes());
+		long contentLength = 0;
+		if(mContent != null) {
+			contentLength = mContent.length();
+		}
+		output.write(("" + HttpRequestHeader.HEADER_CONTENT_LENGTH + ": "
+				+ contentLength + CRLF).getBytes());
+		for (HttpRequestHeader header : mHeaders) {
+			header.encode(output);
+		}
+		output.write(CRLF.getBytes());
+
+		byte[] buffer = new byte[100];
+		while (contentLength!=0) {
+			int len = mContent.read(buffer);
+			if (len < 0) {
+				break;
+			}
+			output.write(buffer, 0, len);
+		}
+		output.flush();
+	}
+
+	// Status-Line = HTTP-Version SP Status-Code SP Reason-Phrase CRLF
+	public static HttpResponse decode(MarkableReader reader, boolean isReadBody) throws IOException {
+		String httpVersion = "";
+		String statusCode = "";
+		String reasonPhrase = "";
+
+		httpVersion = _httpVersion(reader);
+		_sp(reader);
+		statusCode = _statusCode(reader);
+		_sp(reader);
+		reasonPhrase = _reasonPhrase(reader);
+		_crlf(reader);
+		HttpResponse ret = new HttpResponse(httpVersion, statusCode, reasonPhrase);
+
+		
+		try {
+			while (true) {
+				if (isCrlf(reader)) {
+					break;
+				}
+				ret.addHeader(HttpRequestHeader.decode(reader));
+			}
+		} catch (IOException e) {
+		}
+
+		_crlf(reader);
+
+		long size = ret.getContentSizeFromHeader();
+		if(size <0) {size = Integer.MAX_VALUE;}
+		// 
+		// todo MarkableReader内の VFを使用してもよいかも
+		CashKyoroFile vf = new CashKyoroFile(512, 2);
+		if(isReadBody) {
+			int added = 0;
+			byte[] buffer = new byte[1];
+			while(true) {
+				int value = reader.read();
+				if(value<0||added>=size){
+					break;
+				}
+				buffer[0] = (byte)(0xFF&value);
+				vf.addChunk(buffer);
+				added++;
+			}
+		}
+		ret.setContent(vf);
+		return ret;
+	}
+
+	public static String _httpVersion(MarkableReader reader) throws IOException {
+		try {
+			return new String(MarkableReaderHelper.jumpAndGet(reader, available, 256));
+		} catch(IOException e) {
+			throw e;
+		}
+	}
+
+	public static String _statusCode(MarkableReader reader) throws IOException {
+		try {
+			return new String(MarkableReaderHelper.jumpAndGet(reader, available, 256));
+		} catch(IOException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)